client/my-sites/site-settings/settings-section-header.tsx

```diff
@@ -0,0 +1,39 @@
+import { Button, ButtonProps } from '@automattic/components';
+import { useTranslate } from 'i18n-calypso';
+import { ReactNode } from 'react';
+import SectionHeader from 'calypso/components/section-header';
+
+interface SettingsSectionHeaderProps {
+	title: string | ReactNode;
+	id?: string;
+	isSaving?: boolean;
+	disabled?: boolean;
+	showButton?: boolean;
+	onButtonClick?: () => void;
+	buttonProps?: Partial< ButtonProps >;
+	children?: ReactNode;
+}
+
+export default function SettingsSectionHeader( {
+	title,
+	id,
+	isSaving,
+	disabled,
+	showButton,
+	onButtonClick,
+	buttonProps,
+	children,
+}: SettingsSectionHeaderProps ) {
+	const translate = useTranslate();
+
+	return (
+		<SectionHeader label={ title } id={ id }>
+			{ children }
+			{ showButton && (
+				<Button compact primary disabled={ disabled } onClick={ onButtonClick } { ...buttonProps }>
```

Comment:
It's probably okay to ignore in this context, but just fyi the `onButtonClick` prop could be "merged" with the outer one (`buttonProps.onClick`) to avoid losing the internal handler if passed.